crypto/evp/digest.c

```diff
@@ -662,10 +662,7 @@ int EVP_MD_CTX_ctrl(EVP_MD_CTX *ctx, int cmd, int p1, void *p2)
         return 0;
     }
 
-    if (ctx->digest->prov == NULL
-        || (ctx->pctx == NULL
-            || (ctx->pctx->operation != EVP_PKEY_OP_VERIFYCTX
-                && ctx->pctx->operation != EVP_PKEY_OP_SIGNCTX)))
```

Comment:
@mattcaswell - can you confirm this change is good?

Reply:
I think it should be ok

Reply:
Leaving this code in means that some things stop working, for example the operation codes `EVP_PKEY_OP_SIGN` and `EVP_PKEY_OP_VERIFY` (they are perfectly legitimate in the legacy cases)